src/Microsoft.DotNet.Tools.Builder/CompileContext.cs

```diff
@@ -62,7 +60,7 @@ public bool Compile(bool incremental)
                 {
                     var dependencyProjectContext = ProjectContext.Create(dependency.Path, dependency.Framework);
 
-                    if (!NeedsRebuilding(dependencyProjectContext, new ProjectDependenciesFacade(dependencyProjectContext, _args.ConfigValue)))
```

Comment:
If you did the rename, can you please make the DependencyNeedsRebuilding method to accept the ProjectDependency from the current iteration (`dependency`) and then move the construction the dependency projectcontext and dependency dependenciesfacade inside DependencyNeedsRebuilding?